src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -795,5 +797,78 @@ private static BuildVerJson GetBuildDetails()
             }
         }
 
+        private static void ValidateIfTemplateExists(CanvasDocument app, BlockNode node, ErrorContainer errors)
+        {
+            // parse all children of the root nodes to see if al of them have the temeplate they require.
+            foreach(var child in node.Children)
+            {
+                if (string.IsNullOrEmpty(child?.Name?.Kind?.TypeName))
```

Comment:
under what circumstances are child, name, kind, or typename null? As far as I can tell, they should all always have values

Reply:
they should never be null I haven't had any instance of this, just a safe check.

Reply:
have removed these checks since this should never happen, and if it happens there is something wrong and our tool should fail.